docs: 6.0.0 blog post (#2799)
electron/electronjs.org: data/blog/electron-6-0.md

Folks using `electron-osx-sign` to codesign their Electron app shouldn't have to make any changes to their build logic.
If you're codesigning your app with custom scripts, you should ensure
that the two new Helper applications are correctly codesigned.

In order to package your application correctly with these two new helpers you need to be using `electron-packager@14.0.3` or higher.  If you are using `electron-builder` you should follow [this issue](https://github.com/electron-userland/electron-builder/issues/4104) to track support for these new helpers.

## Breaking Changes

 * This release begins laying the groundwork for a future requirement that native Node modules loaded in the renderer process be either [N-API](https://nodejs.org/api/n-api.html) or [Context Aware](https://nodejs.org/api/addons.html#addons_context_aware_addons). The reasons for this change are faster performance, stronger security, and reduced maintenance workload. Read the full details including the proposed timeline in [this issue](https://github.com/electron/electron/issues/18397). This change is expected to be completed in Electron v11.

>  * `http.IncomingMessage` headers have [changed slightly](https://github.com/electron/electron/pull/17517#issue-263752903) to more closely match [Node.js behavior](https://nodejs.org/api/http.html#http_message_headers), particularly with the value of `set-cookie` and how duplicate headers are handled. [#17517](https://github.com/electron/electron/pull/17517).

Reviewer: `net.IncomingMessage` right? we didn't touch the `http` module here

Reviewer 2: good catch. fixed 2765843

￼
￼
￼
￼
￼
 Lock conversation

 * session.clearHostResolverCache() [#17229](https://github.com/electron/electron/pull/17229)
 * session.clearStorageData() [#17249](https://github.com/electron/electron/pull/17249)
 * session.getBlobData() [#17303](https://github.com/electron/electron/pull/17303)
 * session.getCacheSize()  [#17185](https://github.com/electron/electron/pull/17185)
 * session.resolveProxy() [#17222](https://github.com/electron/electron/pull/17222)
 * session.setProxy()  [#17222](https://github.com/electron/electron/pull/17222)
 * webContents.hasServiceWorker() [#16535](https://github.com/electron/electron/pull/16535)
 * webContents.printToPDF() [#16795](https://github.com/electron/electron/pull/16795)
 * webContents.savePage() [#16742](https://github.com/electron/electron/pull/16742)
 * webFrame.executeJavaScript() [#17312](https://github.com/electron/electron/pull/17312)
 * webFrame.executeJavaScriptInIsolatedWorld() [#17312](https://github.com/electron/electron/pull/17312)
 * webviewTag.executeJavaScript() [#17312](https://github.com/electron/electron/pull/17312)

These functions now have two forms, synchronous and Promise-based asynchronous:

>  * `dialog.showMessageBox()` [#17298](https://github.com/electron/electron/pull/17298)

Reviewer: ```suggestion
 * `dialog.showMessageBox()`/`dialog.showMessageBoxSync()` [#17298](https://github.com/electron/electron/pull/17298)
```

>  * `dialog.showOpenDialog()` [#16973](https://github.com/electron/electron/pull/16973)

Reviewer: ```suggestion
 * `dialog.showOpenDialog()`/`dialog.showOpenDialogSync()` [#16973](https://github.com/electron/electron/pull/16973)
```

>  * `dialog.showSaveDialog()` [#17054](https://github.com/electron/electron/pull/17054)

Reviewer: ```suggestion
 * `dialog.showSaveDialog()`/`dialog.showSaveDialogSync()` [#17054](https://github.com/electron/electron/pull/17054)
```

These functions now return Promises:
 * `app.dock.show()` [#16904](https://github.com/electron/electron/pull/16904)

> ### `Electron Helper (Render).app` and `Electron Helper (Plugin).app`

Reviewer: We should add a new block here indicating how folks should support these helpers when packaging .  Something like

```
In order to package your application correctly with these two new helpers you need to be using `electron-packager@14.0.3` or higher.  If you are using `electron-builder` you should follow [this issue](https://github.com/electron-userland/electron-builder/issues/4104) to track support for these new helpers.
```

> There are now more Mac Helper application bundles.

Reviewer: This line is not required

In order to enable the [hardened runtime](https://developer.apple.com/documentation/security/hardened_runtime_entitlements?language=objc), which restricts things like
writable-executable memory and loading code signed by a different Team

> ID, special code signing entitlements must be granted to the Helper.

Reviewer: ```suggestion
ID, special code signing entitlements needed to be granted to the Helper application (`Electron Helper.app`).
```

> To keep the capabilities scoped to the process types that require them, Chromium [added](https://chromium-review.googlesource.com/c/chromium/src/+/1627456) 

Reviewer: ```suggestion
To keep these entitlements scoped to the process types that require them, Chromium [added](https://chromium-review.googlesource.com/c/chromium/src/+/1627456) 
```

two new variants of the Helper app: one for renderers (`Electron Helper (Render).app`) and one for plugins (`Electron Helper (Plugin).app`).

> Currently all three Helpers (the pre-existing `Electron Helper.app` in addition to the two new ones) will be code signed in the same way without

Reviewer: This block of text is inaccurate at this time, we don't code sign these helpers users do.  Something like this block would probably fit better for Electron.

```
Folks using `electron-osx-sign` to codesign their Electron app
shouldn't have to make any changes to their build logic, if you
are code signing your app with custom scripts you should ensure
that the two new Helper applications are correctly codesigned.
```

any entitlements, although this may change in the future.

## Breaking Changes

### `app.allowRendererProcessReuse` option

> This release starts laying the groundwork for a future requirement that native Node modules loaded in the renderer process be either N-API or Context Aware. The reasons for this change are faster performance, stronger security, and reduced maintenance workload. Read the full details in [this issue](https://github.com/electron/electron/issues/18397). This change is expected to complete in Electron v11.

Reviewer: ```suggestion
This release starts laying the groundwork for a future requirement that native Node modules loaded in the renderer process be either [N-API](https://nodejs.org/api/n-api.html) or [Context Aware](https://nodejs.org/api/addons.html#addons_context_aware_addons). The reasons for this change are faster performance, stronger security, and reduced maintenance workload. Read the full details including the proposed timeline in [this issue](https://github.com/electron/electron/issues/18397). This change is expected to be completed in Electron v11.
```

> Outline for change:

Reviewer: This timeline is in the linked issue I don't think we should duplicate it here in case it needs to change

  * Electron 6: Add option to disable site instance patches [#18396](https://github.com/electron/electron/pull/18396)
  * Electron 7: Show deprecation warnings for non-context aware native modules
  * Electron 8: Show deprecation warnings for the default value of `app.allowRendererProcessReuse` to switch
  * Electron 9: Switch the default value of `app.allowRendererProcessReuse`
  * Electron 10: Deprecate the ability to change `app.allowRendererProcessReuse`
  * Electron 11: Remove the ability to change `app.allowRendererProcessReuse`

### Other Breaking Changes

[TODO: need to provide details for the following bullet items or omit from blog post]

  * Fixed disparity between `net` module headers and Node.js'  `http.IncomingMessage` headers. [#17517](https://github.com/electron/electron/pull/17517)
  * Made `ShowItemInFolder` asynchronous with no return value. [#17121](https://github.com/electron/electron/pull/17121)
  * Made app log directory creation opt-in with a new function `app.setAppLogsPath`. [#17841](https://github.com/electron/electron/pull/17841)

## Deprecations

> ## End of Support for 3.0.x

Reviewer: ```suggestion
## End of Support for 3.x.y
```

> Per our [support policy](https://electronjs.org/docs/tutorial/support#supported-versions), 3.0.x has reached end of life. Developers and applications are encouraged to upgrade to a newer version of Electron.

Reviewer: ```suggestion
Per our [support policy](https://electronjs.org/docs/tutorial/support#supported-versions), 3.x.y has reached end of life. Developers and applications are encouraged to upgrade to a newer version of Electron.
```

---
title: Electron 6.0.0
author: 
date: '2019-07-30'
---

> The Electron team is excited to announce the release of Electron 6.0.0! You can install it with npm via `npm install electron@latest` or download it from our [releases website](https://electronjs.org/releases/stable). The release is packed with upgrades, fixes, and new features. We can't wait to see what you build with them! Continue reading for details about this release, and please share any feedback you have!

Reviewer 2: > We can't wait to see what you build with them!

This seems like tradition now :green_heart:

Author: credits to @codebytere from v3.0.0 ❤️

---
title: Electron 6.0.0
author: 
- sofianguy
date: '2019-07-30'
---

The Electron team is excited to announce the release of Electron 6.0.0! You can install it with npm via `npm install electron@latest` or download it from our [releases website](https://electronjs.org/releases/stable). The release is packed with upgrades, fixes, and new features. We can't wait to see what you build with them! Continue reading for details about this release, and please share any feedback you have!

---

## What's New

> This is the first Electron stable release to ever release on the *same day* as [Chrome stable](https://www.chromestatus.com/features/schedule) 🎉

Reviewer: ```suggestion
This is the first Electron stable release to ever release on the *same day* as the corresponding [Chrome stable](https://www.chromestatus.com/features/schedule) 🎉
```

Much of Electron's functionality is provided by the core components of Chromium, Node.js, and V8. Electron keeps up-to-date with these projects to provide our users with new JavaScript features, performance improvements, and security fixes. Each of these packages has a major version bump in Electron 6:

> - Chromium `76.0.3809.77`

Reviewer: ```suggestion
- Chromium `76.0.3809.80`
```

  - [New in 74](https://developers.google.com/web/updates/2019/04/nic74)
  - [New in 75](https://developers.google.com/web/updates/2019/06/nic75)
  - [New in 76](https://developers.google.com/web/updates/tags/chrome76)
- Node.js `12.4.0`
  - [Node 12.4.0 blog post](https://nodejs.org/en/blog/release/v12.4.0/)
- V8 `7.6.303.22`
    - [V8 7.6 blog post](https://v8.dev/blog/v8-release-76)

Electron 6 also includes improvements to Electron-specific APIs. Highlights of the major changes is below; for the full list of changes, check out the [Electron v6.0.0 release notes](https://github.com/electron/electron/releases/tag/v6.0.0).

### Promisification

> Electron 6 continues [Promisification initiative](https://github.com/electron/electron/blob/5-0-x/docs/api/promisification.md) initiative to convert Electron's callback-based API to use Promises. These APIs were converted for Electron 6:

Reviewer: continues and completes?

cc @codebytere can you confirm that all the promisification work is done and in the `6-0-x` branch

Author: Resolved via Slack. @codebytere says promisification are not done yet in v6.

* Cookies API
* `app.dock.show()`
* `contentTracing.getCategories()`
* `contentTracing.getTraceBufferUsage()`
* `contents.executeJavaScript()`
* `webFrame.executeJavaScript()`
* `webFrame.executeJavaScriptInIsolatedWorld()`
* `webviewTag.executeJavaScript()`
* `dialog.showCertificateTrustDialog()`
* `netLog.stopLogging()`
* `session.clearAuthCache()`
* `session.clearHostResolverCache()`
* `session.clearStorageData()`
* `session.getBlobData()`
* `session.getCacheSize()`
* `session.clearCache()`
* `session.setProxy()`
* `session.resolveProxy()`
* `webContents.hasServiceWorker()`
* `webContents.printToPDF()`
* `webContents.savePage()`
* `dialog.showMessageBox()`
* `dialog.showOpenDialog()`
* `dialog.showSaveDialog()`

### `Electron Helper (Render).app` and `Electron Helper (Plugin).app`
[TODO: fill in details about new Helpers]

## Breaking Changes

### `app.allowRendererProcessReuse` option

> Groundwork for requiring any native Node modules that are loaded in the renderer process to be either NAPI or Context Aware. The main reasons for this change are for app performance, security benefits, and lessen maintenance. Read the full details in [this issue](https://github.com/electron/electron/issues/18397). This change is expected to complete in Electron v11.

Reviewer 3: ```suggestion
Groundwork for requiring any native Node modules that are loaded in the renderer process to be either N-API or Context Aware. The main reasons for this change are for app performance, security benefits, and lessen maintenance. Read the full details in [this issue](https://github.com/electron/electron/issues/18397). This change is expected to complete in Electron v11.
```
